designs/2019-esm-compatibilty/README.md

```diff
@@ -0,0 +1,174 @@
+
+This change is additive. It associates `.cjs` files with JS configurations.
+
+The existing semantics of loading CommonJS configurations for CommonJS-based packages do not change.
+
+### Story 2 - Node Compatibility
+
+In Node, ES module and `.cjs` support roll out together.
+
+This change *only* impacts users who define their package as an ES module.
+
+Existing packages that use the default (i.e., CommonJS module resolution) will be unaffected.
+
+## Alternatives
+
+### Alternative 1 - Suppress the error
+
+**[PR#12333](https://github.com/eslint/eslint/pull/12333)**
+
+The fencing exists to prevent users from creating dual-mode (i.e., both ESM/CJS) packages as interoperability between the two format can cause issues. In the case of ESLint, loading a simple config file from a user-defined package should not cause any issues or side-effects.
+
+Eating the exception is viable solution.
+
+### Alternative 2 - Support dynamic `import()`
+
+Instead of using 'import-fresh' to require the config, import it in a cross-format-compatible way using dynamic `import()`.
+
+There has been and will continue to be a lot of talk about using this approach as the CJS/ESM interoperability story is fleshed out.
+
+For now it presents too many unknowns.
+
+## Open Questions
+
+<!--
+    This section is optional, but is suggested for a first draft.
+
+    What parts of this proposal are you unclear about? What do you
+    need to know before you can finalize this RFC?
+
+    List the questions that you'd like reviewers to focus on. When
+    you've received the answers and updated the design to reflect them, 
+    you can remove this section.
+-->
+
+## Help Needed
+
+<!--
+    This section is optional.
+
+    Are you able to implement this RFC on your own? If not, what kind
+    of help would you need from the team?
+-->
+
+## Frequently Asked Questions
+
+> Why doesn't compatibility w/ ESM-based packages 'just work'?
+
+ESLint reaches across the package boundary to retrieve the user-defined configuration. If the consumer package is ESM-based, the `.js` config file will be seen as an ES module. When the ES module loader encounters `module.exports` (i.e., the CommonJS module specifier) it will throw an error.
+
+> What does this interoperability issue affect?
+
+Only ESM-based packages (i.e., packages with `"type": "module"` defined in package.json).
+
+> What about 3rd-party linting tools that provide their own built-in rule set for ESLint (ex [StandardJS](https://standardjs.com/))?
+
+They aren't impacted by this change. 3rd-party modules define both their code and the ESLint rule set used within the same package boundary.
+
+> What about support for ES module configurations
+
+The scope of this RFC is limited to only ES module compatibility concerns. If there is a desire to add support for ESM-based configurations, it will need to be addressed in another RFC.
+
+## Related Discussions
+
+- [Issue #12321](https://github.com/eslint/eslint/pull/12321)
```

Comment:
I'm confused- this seems to be going to the same place as the next line. Am I missing something?

Reply:
Oops, the issue # was wrong to begin with. Updated to the correct # in Rev 3.